src/Authorization/Services/Implementation/ProfileWrapper.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Platform.Authorization.Clients;
+using Altinn.Platform.Authorization.Services.Interface;
+using Altinn.Platform.Profile.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Altinn.Platform.Authorization.Services.Implementation
+{
+    /// <summary>
+    /// Wrapper for the profile api
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class ProfileWrapper : IProfile
+    {
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+        private readonly ProfileClient _profileClient;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProfileWrapper"/> class
+        /// </summary>
+        /// <param name="profileClient">the client handler for profile api in Bridge</param>
+        /// <param name="logger">The logger</param>
+        public ProfileWrapper(ProfileClient profileClient, ILogger<ProfileWrapper> logger)
+        {
+            _profileClient = profileClient;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task<UserProfile> GetUserProfile(int userId)
```

Comment:
Is this something that could be cached?

Reply:
It's currently being cached in the ContextHandler.GetSsnForUser where also the role, oedRole, keyRoleUnit caching etc. is located

Reply:
I do agree though that the caching should be the responsibility of the service/wrapper implementation and not the consumer